File: processor/src/main/java/com/github/snowdream/toybricks/processor/Handler.java
package com.github.snowdream.toybricks.processor;

/**
 * Created by snowdream on 17/2/12.
 */

import javax.annotation.processing.RoundEnvironment;

/**
 * Description: The main base-level handler for performing some action
 */
interface Handler {

    /**
     * Called when the process of the [DBFlowProcessor] is called

     * @param processorManager The manager that holds processing information
     * *
     * @param roundEnvironment The round environment
     */
    void handle(ProcessorManager processorManager, RoundEnvironment roundEnvironment);
}

File: processor/src/main/kotlin/com/github/snowdream/toybricks/processor/Handler.kt
package com.github.snowdream.toybricks.processor

/**
 * Created by snowdream on 17/2/12.
 */

import javax.annotation.processing.RoundEnvironment

/**
 * Description: The main base-level handler for performing some action
 */
internal interface Handler {

    /**
     * Called when the process of the [ToyBricksProcessor] is called

     * @param processorManager The manager that holds processing information
     * * *
     * *
     * @param roundEnvironment The round environment
     */
    fun handle(processorManager: ProcessorManager, roundEnvironment: RoundEnvironment)
}